pynxtools_mpes/reader.py

```diff
@@ -285,128 +163,100 @@ def _getattr(obj, attr):
-                precursor = value.split(":")[0]
-                value = value[value.index(":") + 1 :]
-
-                # Filling in the data and axes along with units from xarray
-                if precursor == "@data":
-                    try:
-                        template[key] = rgetattr(
-                            obj=x_array_loaded,
-                            attr=value,
-                        )
-                        if key.split("/")[-1] == "@axes":
-                            template[key] = list(template[key])
-
-                    except ValueError:
-                        print(
-                            f"Incorrect axis name corresponding to " f"the path {key}",
-                        )
-
-                    except AttributeError:
-                        print(
-                            f"Incorrect naming syntax or the xarray doesn't "
-                            f"contain entry corresponding to the path {key}",
-                        )
-
-                # Filling in the metadata from xarray
-                elif precursor == "@attrs":
-                    if key not in eln_data_dict:
-                        try:  # Tries to fill the metadata
-                            template[key] = iterate_dictionary(
-                                x_array_loaded.attrs,
-                                value,
-                            )
-
-                        except KeyError:
-                            print(
-                                f"[info]: Path {key} not found. "
-                                f"Skipping the entry.",
-                            )
-
-                if isinstance(template.get(key), str) and template[key].startswith(
-                    "@link:"
-                ):
-                    template[key] = {"link": template[key][6:]}
-            else:
-                # Fills in the fixed metadata
-                template[key] = value
-
-        # Filling in ELN metadata and overwriting the common paths by
-        # giving preference to the ELN metadata
-        for key, value in eln_data_dict.items():
-            template[key] = value
-
-        # remove groups that have required children missing
-        for key in optional_groups_to_remove:
-            if template.get(key) is None:
-                group_to_delete = key.rsplit("/", 1)[0]
-                logger.info(
-                    f"[info]: Required element {key} not provided. "
-                    f"Removing the parent group {group_to_delete}.",
-                )
-                for temp_key in template.keys():
-                    if temp_key.startswith(group_to_delete):
-                        del template[temp_key]
-
-        return template
+    def get_attr(self, key: str, path: str) -> Any:
+        try:
+            return iterate_dictionary(self.data_xarray.attrs, path)
+        except KeyError:
+            logger.info(f"Path {path} not found. Skipping the entry.")
```

Comment:
If the key is not found in attrs, but maybe later in eln, this info is still issued, no? Isn't there a corresponding warning also in the multi reader, that collects all these?